### PR TITLE
SQL: Fix DATETIME_PARSE behaviour regarding timezones

### PR DESCRIPTION
**Pull request by matriv**
Previously, when the timezone was missing from the datetime string
and the pattern, UTC was used, instead of the session defined timezone.
Moreover, if a timezone was included in the datetime string and the
pattern then this timezone was used. To have a consistent behaviour
the resulting datetime will always be converted to the session defined
timezone, e.g.:
```
SELECT DATETIME_PARSE('2020-05-04 10:20:30.123 +02:00', 'HH:mm:ss dd/MM/uuuu VV') AS datetime;
```
with `time_zone` set to `-03:00` will result in
```
2020-05-04T05:20:40.123-03:00
```

Follows: #54960

### CODE REVIEW

File: x-pack/plugin/sql/src/main/java/org/elasticsearch/xpack/sql/expression/function/scalar/datetime/DateTimeParseProcessor.java [hunk 1]
```diff
@@ -55,9 +53,9 @@ public static Object process(Object timestampStr, Object pattern) {
             TemporalAccessor ta = DateTimeFormatter.ofPattern((String) pattern, Locale.ROOT)
                 .parseBest((String) timestampStr, ZonedDateTime::from, LocalDateTime::from);
             if (ta instanceof LocalDateTime) {
-                return ZonedDateTime.ofInstant((LocalDateTime) ta, ZoneOffset.UTC, UTC);
+                return ZonedDateTime.ofInstant((LocalDateTime) ta, zoneId.getRules().getValidOffsets(((LocalDateTime) ta)).get(0), zoneId);
```
Comment (costin) [inline x-pack/plugin/sql/src/main/java/org/elasticsearch/xpack/sql/expression/function/scalar/datetime/DateTimeParseProcessor.java:56]: The `get(0)` seems optimistic - please extract this into a (date) utils method (maybe there's some utility in ES we could use).

**Review by costin — APPROVED** [review]
LGTM

File: docs/reference/sql/functions/date-time.asciidoc [hunk 2]
```diff
@@ -487,6 +484,16 @@ include-tagged::{sql-specs}/docs/docs.csv-spec[dateTimeParse1]
 include-tagged::{sql-specs}/docs/docs.csv-spec[dateTimeParse2]
 --------------------------------------------------
 
+[NOTE]
+====
+If timezone is not specified in the datetime string expression and the parsing pattern, the resulting `datetime` will have the
+time zone specified by the user through the `time_zone`/`timezone` REST/driver parameters with no conversion applied.
```
Comment (astefan) [inline docs/reference/sql/functions/date-time.asciidoc:490]: As an improvement, I would add links to REST and jdbc driver params documentation where these parameters are defined.

**Review by astefan — APPROVED** [review]
LGTM. Left one suggestion.

**Pull request merged**